folding-schemes/src/frontend/circom/mod.rs

```diff
@@ -250,4 +302,34 @@ pub mod tests {
             .unwrap();
         assert_eq!(z_i1_var.value().unwrap(), vec![Fr::from(52u32)]);
     }
+
+    #[test]
+    fn test_custom_code() {
+        let r1cs_path = PathBuf::from("./src/frontend/circom/test_folder/cubic_circuit.r1cs");
+        let wasm_path =
+            PathBuf::from("./src/frontend/circom/test_folder/cubic_circuit_js/cubic_circuit.wasm");
+
+        let circom_fcircuit = CircomFCircuit::<Fr>::new((r1cs_path, wasm_path, 1, 0)).unwrap(); // state_len:1, external_inputs_len:0
+
+        let new_circuit = circom_fcircuit.set_custom_code(Rc::new(move |_i, z_i, _external| {
```

Comment:
Any reason to using `move`? Since all the required parameters are passed to the closure.

```suggestion
        let new_circuit = circom_fcircuit.set_custom_code(Rc::new(|_i, z_i, _external| {
```

Reply:
solved it